### PR TITLE
Grunt setup and tasks improvements

### PR DESCRIPTION
**Pull request by matiasbeckerle**
Could you review it guys? Please take a deep breath. I know there is a lot of things to improve because this changes are faaar away from being "the ones". I also put some TODOs at the beginning of the Gruntfile. 

For this PR I will ask only for errors/issues. If you see something to improve then I will add it to the TODOs list and develop in another branch, another time, another PR. 

You could find some packages updates too.

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -50,21 +50,10 @@
...
-        <script src="/modules/base/services/authService.js"></script>
-        <script src="/modules/base/services/menus.js"></script>
-        <script src="modules/base/directives/confirmMessage.js"></script>
-        <script src="modules/base/services/authService.js"></script>
-        <script src="modules/base/services/menus.js"></script>
-        <script src="modules/base/interceptors/authInterceptor.js"></script>
-        <!-- //===== mean-cli hook =====// -->
-
+        <!-- App scripts -->
+        <script src="/dist/scripts/all.min.js"></script>
```
Comment (mravinale) [inline public/index.html]: not sure, I think we should work by default without the minification, in dev mode, the minification should happen when explicitly want to build using something like: `grunt prod`

File: README.md [hunk 2]
```diff
@@ -16,18 +16,34 @@ A bootstrapp application based on MongoDB, Express, Angular, Node, Passport.
...
 
 Go to [MongoDB official downloads](http://www.mongodb.org/downloads) and follow the [instructions](http://docs.mongodb.org/manual/installation/)
 
-### Running the app for development
+### Development
 
-Run ```grunt dev``` to start the app in development mode with client-server livereload. (Remember to have the mongoDB running)
+We are currently using Grunt as our task runner. For general development only run:
+
+`grunt`
```
Comment (mravinale) [inline README.md]: I think we need `grunt dev` for development without minification nad with full reload support etc.. `grunt pord` making the replacements on index page with the minified vesion of js and css in a vfolder named: `dist` or `build ` whatever

File: Gruntfile.js [hunk 3]
```diff
@@ -110,71 +180,103 @@ module.exports = function (grunt) {
...
+    /**
+     * Default task will be our dev task
+     */
+    grunt.registerTask('default', [
         'express:dev',
-        'open',
-        'watch'
-    ]);
-
-    grunt.registerTask('ui-dev', [
```
Comment (mravinale) [inline Gruntfile.js]: I think we need some grunt ui-dev specifc for `ui developers`

Comment (matiasbeckerle) [inline Gruntfile.js]: Agree. Is already in the TODO list at the beginning of this file.

File: README.md (hunk 2 above)
Comment (matiasbeckerle) [inline README.md]: The creation of a task for building is already in the TODO list at the beginning of the file.

File: public/index.html (hunk 1 above)
Comment (matiasbeckerle) [inline public/index.html]: So how do you solve the `index.html` file? using a pre-compiling tool? Something like:

```
<!-- if(production) -->
<script src="/dist/scripts/all.min.js"></script>

<!-- if(dev) -->
[all scripts list goes here]
```

**Comment by matiasbeckerle** [comment]
What do you think now @mravinale? Could you take another look?

**Comment by matiasbeckerle** [comment]
Ok I will merge it in order to keep a progress. Please let me know if you disagree with any matter, @mravinale.

**Pull request merged**